contrib/fslhd.py

```diff
@@ -257,8 +260,39 @@ def generate_nifti_fields(header):
         'sform_xorient': ORIENTATION_STRINGS[nib.aff2axcodes(sform)[0]],
         'sform_yorient': ORIENTATION_STRINGS[nib.aff2axcodes(sform)[1]],
         'sform_zorient': ORIENTATION_STRINGS[nib.aff2axcodes(sform)[2]],
-        'file_type': f"NIFTI-{int(nib_fields['magic'].decode('utf-8')[2])}+",
+        'file_type': f"NIFTI-{int(nib_fields['magic'].decode('utf-8', errors='ignore')[2])}+",
         'file_code': int(nib_fields['magic'].decode("utf-8")[2]),
-        'descrip': nib_fields['descrip'].decode('utf-8'),
-        'aux_file': nib_fields['aux_file'].decode('utf-8')
+        'descrip': nib_fields['descrip'].decode("utf-8", errors="ignore"),
+        'aux_file': nib_fields['aux_file'].decode("utf-8", errors="ignore")
+    }
+
+
+def generate_numpy_fields(arr):
+    """
+    Generate useful summary statistics for any arbitrary numpy array.
+    """
+    arr = arr.astype(np.float64)
+    return {
+        'arr_nbytes': arr.nbytes,
+        'arr_size': arr.size,
+        'arr_num_zeros': arr[arr == 0.0].size,
+        'arr_num_nonzeros': arr[arr != 0.0].size,
+        'arr_num_positives': arr[arr > 0.0].size,
+        'arr_num_negatives': arr[arr < 0.0].size,
+        'arr_num_nans': arr[arr == np.nan].size,
+        'arr_num_infs': arr[arr == np.inf].size,
+        'arr_num_unique': np.unique(arr).size,
+        'arr_min': arr.min(),
+        'arr_max': arr.max(),
+        'arr_median': np.median(arr),
+        'arr_mode': mode(arr, axis=None, keepdims=False).mode,
+        'arr_mean': arr.mean(),
+        'arr_std': arr.std(),
+        'arr_sum': arr.sum(),
+        'arr_p1': np.percentile(arr, 1),
+        'arr_p10': np.percentile(arr, 10),
+        'arr_p90': np.percentile(arr, 90),
+        'arr_p99': np.percentile(arr, 99),
+        'arr_iqr': np.percentile(arr, 75) - np.percentile(arr, 25),
+        'arr_mad': np.median(np.abs(arr - np.median(arr))),
```

Comment:
We probably don't need all of these to tell if the array has changed, hah...

Reply:
TIL about [median absolute deviation](https://en.wikipedia.org/wiki/Median_absolute_deviation), I'd never heard of it before.